[CI] ApproximationGoldenTests testApproximationDisabled failing

**Build Scans:**
- [elasticsearch-intake #38162 / part3](https://gradle-enterprise.elastic.co/s/pildlwi7q7mcs)
- [elasticsearch-pull-request #139351 / part-3](https://gradle-enterprise.elastic.co/s/ytytsxgyx2id6)
- [elasticsearch-pull-request #139351 / part-3](https://gradle-enterprise.elastic.co/s/j4wghbgls3o5q)
- [elasticsearch-pull-request #139350 / part-3](https://gradle-enterprise.elastic.co/s/pedwbttkfgzo6)
- [elasticsearch-intake #38157 / part3](https://gradle-enterprise.elastic.co/s/els5xdpii5tx4)
- [elasticsearch-intake #38158 / part3](https://gradle-enterprise.elastic.co/s/k2jvhbcvjssqo)
- [elasticsearch-pull-request #139350 / part-3](https://gradle-enterprise.elastic.co/s/rp52ngdj6cydk)
- [elasticsearch-pull-request #139339 / part-3](https://gradle-enterprise.elastic.co/s/xjrwmx7dlin52)
- [elasticsearch-pull-request #139340 / part-3](https://gradle-enterprise.elastic.co/s/ab5qtshowvun2)
- [elasticsearch-intake #38161 / part3](https://gradle-enterprise.elastic.co/s/bmpqy4kqzfkd4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.optimizer.ApproximationGoldenTests.testApproximationDisabled" -Dtests.seed=A0E1A86080D40259 -Dtests.locale=es-AR -Dtests.timezone=Brazil/West -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.optimizer.ApproximationGoldenTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testApproximationDisabled'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Output for test 'testApproximationDisabled' does not match for stages '[LOGICAL_OPTIMIZATION, PHYSICAL_OPTIMIZATION, LOCAL_PHYSICAL_OPTIMIZATION]'
```

**Issue Reasons:**
- [main] 14 consecutive failures in test testApproximationDisabled
- [main] 5 consecutive failures in step part3
- [main] 8 consecutive failures in step part-3
- [main] 1 consecutive failures in step part-3-fips-140-3
- [main] 2 consecutive failures in step checkpart3_gpu-tests
- [main] 5 consecutive failures in pipeline elasticsearch-intake
- [main] 4 consecutive failures in pipeline elasticsearch-pull-request
- [main] 33 failures in test testApproximationDisabled (20.2% fail rate in 163 executions)
- [main] 6 failures in step part3 (10.0% fail rate in 60 executions)
- [main] 24 failures in step part-3 (25.8% fail rate in 93 executions)
- [main] 2 failures in step checkpart3_gpu-tests (100.0% fail rate in 2 executions)
- [main] 6 failures in pipeline elasticsearch-intake (10.0% fail rate in 60 executions)
- [main] 13 failures in pipeline elasticsearch-pull-request (16.5% fail rate in 79 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -364,9 +364,6 @@ tests:
 - class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
   method: testBulkOperations {useLegacyFormat=false useSyntheticSource=false}
   issue: https://github.com/elastic/elasticsearch/issues/147130
-- class: org.elasticsearch.xpack.esql.optimizer.ApproximationGoldenTests
-  method: testApproximationDisabled
-  issue: https://github.com/elastic/elasticsearch/issues/147144
 - class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
   method: testReplaceChildren {class org.elasticsearch.xpack.esql.plan.physical.ExchangeSinkExec}
   issue: https://github.com/elastic/elasticsearch/issues/147164

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java
@@ -74,6 +74,7 @@ import org.elasticsearch.xpack.esql.analysis.EnrichResolution;
 import org.elasticsearch.xpack.esql.analysis.MutableAnalyzerContext;
 import org.elasticsearch.xpack.esql.analysis.UnmappedResolution;
 import org.elasticsearch.xpack.esql.analysis.Verifier;
+import org.elasticsearch.xpack.esql.approximation.ApproximationSettings;
 import org.elasticsearch.xpack.esql.core.expression.Alias;
 import org.elasticsearch.xpack.esql.core.expression.Attribute;
 import org.elasticsearch.xpack.esql.core.expression.Expression;
@@ -708,7 +709,7 @@ public final class EsqlTestUtils {
             AnalyzerSettings.QUERY_TIMESERIES_RESULT_TRUNCATION_MAX_SIZE.getDefault(Settings.EMPTY),
             AnalyzerSettings.QUERY_TIMESERIES_RESULT_TRUNCATION_DEFAULT_SIZE.getDefault(Settings.EMPTY),
             null,
-            statement.setting(QuerySettings.APPROXIMATION),
+            new ApproximationSettings.Builder(false).merge(statement.setting(QuerySettings.APPROXIMATION)).build(),
             Map.of()
         );
     }

